src/diffusers/utils/hub_utils.py

```diff
@@ -129,3 +131,67 @@ def create_model_card(args, model_name):
 
     card_path = os.path.join(args.output_dir, "README.md")
     model_card.save(card_path)
+
+
+# Old default cache path, potentially to be migrated.
+# This logic was more or less taken from `transformers`, with the following differences:
+# - Diffusers doesn't use custom environment variables to specify the cache path.
+# - There is no need to migrate the cache format, just move the files to the new location.
+hf_cache_home = os.path.expanduser(
+    os.getenv("HF_HOME", os.path.join(os.getenv("XDG_CACHE_HOME", "~/.cache"), "huggingface"))
+)
+old_diffusers_cache = os.path.join(hf_cache_home, "diffusers")
+
+
+def move_cache(old_cache_dir=None, new_cache_dir=None):
+    if new_cache_dir is None:
+        new_cache_dir = DIFFUSERS_CACHE
+    if old_cache_dir is None:
+        old_cache_dir = old_diffusers_cache
+
+    for file in os.listdir(old_cache_dir):
+        # Move directories only
+        if os.path.isdir(os.path.join(old_cache_dir, file)):
+            # Skip directories that already exist in the new cache
+            if os.path.isdir(os.path.join(new_cache_dir, file)):
+                logger.warning(
+                    f"Skipping migration of directory {file} because it already exists in the new cache location."
+                )
+            shutil.move(os.path.join(old_cache_dir, file), os.path.join(new_cache_dir, file))
```

Comment:
This is much simpler than https://github.com/huggingface/transformers/blob/a45914193ab72da62bb7eaae524eec5a7543c8f1/src/transformers/utils/hub.py#L1033 because I believe we don't need to perform any migration (other than moving the model directories to the new location). Therefore I don't think we need to retrieve metadata from the hub.